### PR TITLE
Ef/task3 dapper

### PR DESCRIPTION
**Pull request by vojtko84**
Ef/task3 dapper

### CODE REVIEW

File: EntityFramework/Delivery/Delivery/Program.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{
+    internal class Program
+    {
+        private static void Main(string[] args)
+        {
+        }
+
+        private static IConfigurationRoot Initialize()
+        {
+            var bilder = new ConfigurationBuilder().
```
Comment (yevhenbilushko) [inline EntityFramework/Delivery/Delivery/Program.cs:14]: builder

File: EntityFramework/Delivery/Delivery/Repositories/DapperContribRepository.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+using Dapper.Contrib.Extensions;
+using Delivery.Models;
+
+namespace Delivery.Repositories
+{
+    public class DapperContribRepository : IRepository
```
Comment (yevhenbilushko) [inline EntityFramework/Delivery/Delivery/Repositories/DapperContribRepository.cs:10]: better "CategoryRepository"

File: EntityFramework/Delivery/Delivery/Repositories/DapperContribRepository.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        private readonly IDbConnection db;
```
Comment (yevhenbilushko) [inline EntityFramework/Delivery/Delivery/Repositories/DapperContribRepository.cs:12]: _db

File: EntityFramework/Delivery/Delivery/Repositories/DapperContribRepository.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        public int AddCategory(Category category)
+        {
+            var newCategory = db.Insert<Category>(category);
+            return (int)newCategory;
+        }
+
+        public int AddCategoryWithObjects(Category category)
+        {
+            var newCategory = db.Insert<Category>(category);
+            if (category.Products.Count != 0)
```
Comment (yevhenbilushko) [inline EntityFramework/Delivery/Delivery/Repositories/DapperContribRepository.cs:28]: can we use Any ?

Comment (yevhenbilushko) [reply EntityFramework/Delivery/Delivery/Repositories/DapperContribRepository.cs:28]: add add brakets for "If" not to miss logic

File: EntityFramework/Delivery/Delivery/Repositories/DapperRepository.cs [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            var id = categories.LastOrDefault().Id;
+            return id;
+        }
+
+        public int AddCategoryWithObjects(Category category)
+        {
+            var sql = "INSERT INTO [dbo].[Products] ([Name], [Description], [Price], [CategoryId], [ProviderId]) VALUES (@Name, @Description, @Price, @CategoryId, @ProviderId)";
+            AddCategory(category);
+            var categories = db.Query<Category>($"SELECT * FROM [dbo].[Categories] WHERE Name = '{category.Name}'").ToList();
+            var id = categories.LastOrDefault().Id;
```
Comment (yevhenbilushko) [inline EntityFramework/Delivery/Delivery/Repositories/DapperRepository.cs:33]: try to do it by SQL 
i mean get last and select only Id

File: EntityFramework/Delivery/Delivery/Repositories/DapperContribRepository.cs (hunk 4 above)
Comment (vojtko84) [reply EntityFramework/Delivery/Delivery/Repositories/DapperContribRepository.cs:28]: I don't understand why to use Any, i don't check collection items. Or have I misunderstood something?

File: EntityFramework/Delivery/Delivery/Repositories/DapperCategoryRepository.cs [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+            var id = categories.LastOrDefault().Id;
+            return id;
+        }
+
+        public int AddCategoryWithObjects(Category category)
+        {
+            var sql = "INSERT INTO [dbo].[Products] ([Name], [Description], [Price], [CategoryId], [ProviderId]) VALUES (@Name, @Description, @Price, @CategoryId, @ProviderId)";
+            AddCategory(category);
+            var categoryId = _db.Query<int>($"SELECT Max(Id) FROM [dbo].[Categories]").FirstOrDefault();
+            if (category.Products.Count != 0)
```
Comment (yevhenbilushko) [inline EntityFramework/Delivery/Delivery/Repositories/DapperCategoryRepository.cs:33]: maybe use "category.Products.Any()" ?

File: EntityFramework/Delivery/Delivery/Repositories/DapperContribRepository.cs (hunk 4 above)
Comment (yevhenbilushko) [reply EntityFramework/Delivery/Delivery/Repositories/DapperContribRepository.cs]: just "Count != 0" can be writen by "Any()"

**Review by yevhenbilushko — APPROVED** [review]

**Comment by yevhenbilushko** [comment]
u can merge it

**Pull request merged**